pkg/ingester/activeseries/active_series.go

```diff
@@ -369,6 +371,8 @@ func (s *seriesStripe) findAndUpdateOrCreateEntryForSeries(ref storage.SeriesRef
 					s.activeMatchingNativeHistograms[match]++
 					s.activeMatchingNativeHistogramBuckets[match] += uint32(numNativeHistogramBuckets)
 				}
+				s.cat.Decrement(series, -1)
```

Comment:
```suggestion
				s.cat.Decrement(series, entry.numNativeHistogramBuckets)
```

Reply:
This would make this block equal in all three cases, so why we don't just move it down where we do `entry.numNativeHistogramBuckets = numNativeHistogramBuckets`?